src/Client/OfficeInterop/OfficeInterop.fs

```diff
@@ -1850,17 +1837,15 @@ let getSelectedCellType () =
                 let! arcTable = ArcTable.tryGetFromExcelTable(excelTable, context)
                 let! (arcMainColumn, _) = getArcMainColumn excelTable arcTable.Value context
 
-                if (arcMainColumn.Validate false) && rowIndex > 0 then
+                if rowIndex > 0 then
 
                     let result =
                         match arcMainColumn with
-                        | amc when amc.Header.ToString() = "Output [Sample Name]" -> None
-                        | amc when amc.Header.ToString() = "Input [Source Name]" -> None
-                        | amc when amc.Cells.[(int rowIndex) - 1].isUnitized -> Some BodyCellType.Unitized
-                        | amc when amc.Cells.[(int rowIndex) - 1].isTerm -> Some BodyCellType.Term
-                        | amc when amc.Cells.[(int rowIndex) - 1].isData -> Some BodyCellType.Data
-                        | amc when amc.Header.isInput && amc.Cells.[(int rowIndex) - 1].isFreeText -> Some BodyCellType.Text
-                        | amc when amc.Header.isOutput && amc.Cells.[(int rowIndex) - 1].isFreeText -> Some BodyCellType.Text
+                        | amc when amc.Cells.[(int rowIndex) - 1].isUnitized -> Some CompositeCellDiscriminate.Unitized
+                        | amc when amc.Cells.[(int rowIndex) - 1].isTerm -> Some CompositeCellDiscriminate.Term
+                        | amc when amc.Cells.[(int rowIndex) - 1].isData -> Some CompositeCellDiscriminate.Data
+                        | amc when amc.Header.isInput && amc.Header.IsDataColumn && amc.Cells.[(int rowIndex) - 1].isFreeText -> Some CompositeCellDiscriminate.Text
```

Comment:
why are you checking headers for this again? Can you not simply check if cell _.isFreetext?